src/app/services/edition-data.service.ts

```diff
@@ -30,11 +32,57 @@ export class EditionDataService {
 
         return editionDoc;
       }),
+      mergeMap((editionData) => this.loadXIinclude(editionData, editionUrl.substring(0, editionUrl.lastIndexOf('/') + 1))),
       publishReplay(1),
       refCount(),
       catchError(() => this.handleLoadingError()));
   }
 
+  loadXIinclude(doc: HTMLElement, baseUrlPath: string) {
+    const filesToInclude = Array.from(doc.getElementsByTagName('xi:include'));
+    const xiIncludeLoadsSubs = [];
+    filesToInclude.forEach(element => {
+      const fileXInclude = baseUrlPath + element.getAttribute('href');
+      const fileXpointer = element.getAttribute('xpointer');
+      xiIncludeLoadsSubs.push(
+        this.http.get(fileXInclude, { responseType: 'text' }).pipe(
+          tap((fileData) => {
+            let includedDoc: XMLElement;
+            if (typeof fileData !== 'object' && typeof fileData === 'string') {
+              includedDoc = parseXml(fileData);
+            } else {
+              includedDoc = fileData;
+            }
+            let includedTextElem;
+            if (fileXpointer) {
+              includedTextElem = doc.querySelector(`[*|id="${fileXpointer}"]`) || includedDoc.querySelector('text');
+            } else {
+              includedTextElem = includedDoc.querySelector('text');
+            }
+            element.parentNode.replaceChild(includedTextElem, element);
```

Comment:
Why do you replace the element? 
Would it be viable to store the xi:included elements elsewhere (a map URL -> XML for example)?

Reply:
IMHO Once you've retrieved the content linked with xi:included you don't need that element anymore. It is used just to split very long documents in smaller and more manageable files, and document with xi:included elements should be considered as a single-file document.